editor/src/components/inspector/sections/layout-section/conditional-section.tsx

```diff
@@ -126,12 +128,18 @@ export const ConditionalSection = React.memo(({ paths }: { paths: ElementPath[]
     'ConditionalSection condition override',
   )
 
-  const conditionExpression = useEditorState(
+  const originalConditionExpression = useEditorState(
     Substores.metadata,
     (store) => conditionExpressionSelector(store, paths),
     'ConditionalSection condition expression',
   )
 
+  const [conditionExpression, setConditionExpression] = React.useState(originalConditionExpression)
+
+  React.useEffect(() => {
+    setConditionExpression(originalConditionExpression)
+  }, [originalConditionExpression])
+
```

Comment:
```suggestion
  const [conditionExpression, setConditionExpression] = usePropControlledStateV2(originalConditionExpression)
```

I believe usePropControlledStateV2 achieves exactly this behavior, but without an extra re-render penalty, and no useEffect

Reply:
oh that's very nice, thank you for the heads up 👌

Reply:
https://github.com/concrete-utopia/utopia/pull/3436/commits/d8d4b5ad806be19b40c4f403e1d23bb2d51a466a